### PR TITLE
UISEES-35: Add suggestions to the “Advanced search” textbox

### PR DESCRIPTION
**Pull request by Dmytro-Melnyshyn**
[UISEES-35: Add suggestions to the “Advanced search” textbox](https://issues.folio.org/browse/UISEES-35)

## Purpose
Add a functionality to `ElasticQueryField` (elastic search) component, regarding UISEES-35 requirements.

## Approach
Added a `TextField` component to `ui-inventory-es` to alter the passed value in the `onChange` method from `event` to `event.target.value`, as I need to transfer the modified event value from the `ElasticQueryField`.

According to the requirements that PO added in the [story](https://issues.folio.org/browse/UISEES-26) comments, if the user enters an expression without structure (search option - operator - term) and presses `enter` we should search by `Keyword (title, contributor, identifier)`. To do this, I added the `isSearchByKeyword` state to the `SearchAndSort` component and the ability to change this state using the `setIsSearchByKeyword` method from the `ElasticQueryField`.

### CODE REVIEW

File: src/components/ElasticQueryField/ElasticQueryField.js [hunk 1]
```diff
@@ -1,111 +1,444 @@
...
-    if (!searchOption) return searchOptions;
-    return [];
+  const resetStructure = () => {
+    setSearchOption('');
+    setOperator('');
+    setTerm('');
+  };
+
+  const addQuotes = valueToInsert => {
+    if (valueToInsert.includes(' ')) {
```
Comment (Godlevskyi) [inline src/components/ElasticQueryField/ElasticQueryField.js]: Please add space, brackets to the constant.

File: src/components/ElasticQueryField/ElasticQueryField.js [hunk 2]
```diff
@@ -1,111 +1,444 @@
...
+        onChange={handleChange}
         onKeyDown={handleKeyDown}
       />
-      {value && renderDropDown()}
+      <p
+        role="alert"
+        id="info"
+        className={css.info}
+      >
+        When autocomplete results are available use up and down arrows to review and enter to select.
```
Comment (Godlevskyi) [inline src/components/ElasticQueryField/ElasticQueryField.js]: Please add it to translation

File: src/components/ElasticQueryField/ElasticQueryField.js (hunk 1 above)
Comment (Dmytro-Melnyshyn) [reply src/components/ElasticQueryField/ElasticQueryField.js]: Thanks for the review, fixed.

File: src/components/ElasticQueryField/ElasticQueryField.js (hunk 2 above)
Comment (Dmytro-Melnyshyn) [reply src/components/ElasticQueryField/ElasticQueryField.js]: Thanks for the review, fixed.

**Review by Godlevskyi — APPROVED** [review]

**Comment by JohnC-80** [comment]
@Dmytro-Melnyshyn If I understand this correctly, you need to conform the parameters of onChange handlers somehow - I recommend using some type of check for the event structure to conform logic at the higher level rather than require the base components to conform to a certain behavior and deliver onChange paramters in a particular way - this would preserve options so that it wouldn't matter how the component delivers its parameters for onChange - something like this that just checks for a few properties from the traditional, web-standard Event : https://github.com/redux-form/redux-form/blob/0485967d7f98b349e850570a2f2f3a4124f392d8/src/events/isEvent.js
So the logic around that can act accordingly to procure the value that it needs.

This can be merged as is for sake of the POC, but I'm not apt to depart from standard onChange event handlers in the basic centralized form components - this would be quite a breaking change in that case.

**Comment by Dmytro-Melnyshyn** [comment]
> @Dmytro-Melnyshyn If I understand this correctly, you need to conform the parameters of onChange handlers somehow - I recommend using some type of check for the event structure to conform logic at the higher level rather than require the base components to conform to a certain behavior and deliver onChange paramters in a particular way - this would preserve options so that it wouldn't matter how the component delivers its parameters for onChange - something like this that just checks for a few properties from the traditional, web-standard Event : https://github.com/redux-form/redux-form/blob/0485967d7f98b349e850570a2f2f3a4124f392d8/src/events/isEvent.js
> So the logic around that can act accordingly to procure the value that it needs.
> 
> This can be merged as is for sake of the POC, but I'm not apt to depart from standard onChange event handlers in the basic centralized form components - this would be quite a breaking change in that case.

@JohnC-80 Thanks for your suggestions, I removed the TextField component and added validation with isEvent. Thank you!

**Pull request merged**